### PR TITLE
util/logutil: Remove useless grpc log in production

### PR DESCRIPTION
**Pull request by SabaPing**
### What problem does this PR solve?

There is a pr https://github.com/pingcap/tidb/pull/23534, refactors the log code. It rewrites the initialization logic of the grpc logger, resulting in annoying grpc logs.

This pr remove those useless logs.

### What is changed and how it works?

What's Changed:

Refactor the grpc logger initialization logic.

### Related changes

https://github.com/pingcap/tidb/pull/23534

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

- No release note

### CODE REVIEW

File: util/logutil/log.go [hunk 1]
```diff
@@ -111,17 +109,64 @@ func InitLogger(cfg *LogConfig) error {
...
+
+func initGRPCLogger(cfg *LogConfig) error {
+
+	config := deepcopyLogConfig(&cfg.Config)
+	// hack: force stdout
+	config.File.Filename = ""
 	if len(os.Getenv("GRPC_DEBUG")) > 0 {
-		// more information for verbosity: https://github.com/google/glog#verbose-logging
-		gzap.ReplaceGrpcLoggerV2WithVerbosity(gl, GRPCLogDebugVerbosity)
+		config.Level = "debug"
```
Comment (breeswish) [inline util/logutil/log.go:126]: how about:

```golang
config := *cfg.Config
config.Level = "debug"
```

In this way we can make sure that `cfg` is not modified as well.

**Review by breeswish — APPROVED** [review]

File: util/logutil/log.go [hunk 2]
```diff
@@ -111,12 +109,33 @@ func InitLogger(cfg *LogConfig) error {
...
+	}
+
+	return nil
+}
+
+func initGRPCLogger(cfg *LogConfig) error {
+	// copy Config struct by assignment
+	config := cfg.Config
+	// hack: force stdout
+	config.File.Filename = ""
```
Comment (breeswish) [inline util/logutil/log.go:124]: Why do we need to force output grpc log to stdout?

Comment (SabaPing) [reply util/logutil/log.go:124]: The old logic directs all grpc log to std. Please take a look at https://github.com/pingcap/tidb/blob/4f31cb49f083870f6b8e0f1ecc9183bc40c4ba42/tidb-server/main.go#L592

Comment (breeswish) [reply util/logutil/log.go:124]: Nice catch. Then there is a problem: the old logic output to `stderr`. In the tiup deployment, stderr will be piped to the `tidb_stderr.log` file eventually. But if you redirect them to the stdout as in this PR, then you will just lose your outputs, as stdout is not collected anywhere..

Comment (breeswish) [reply util/logutil/log.go:124]: If you find it hard to print to stderr then I think it is fine to print in the log file. There is no much difference from my perspective.

File: util/logutil/log_test.go [hunk 3]
```diff
@@ -122,3 +122,28 @@ func (s *testLogSuite) TestSetLevel(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(log.GetLevel(), Equals, zap.DebugLevel)
 }
+
+func (s *testLogSuite) TestGrpcLoggerCreation(c *C) {
+	conf := NewLogConfig("info", DefaultLogFormat, "", EmptyFileLogConfig, false)
+	copiedConfig := *conf
+	// assert that the new conf's address not same as the old one
+	c.Assert(&copiedConfig != conf, Equals, true)
```
Comment (breeswish) [inline util/logutil/log_test.go:130]: how about aseerting that after calling the function the original config's level remains unchanged?

**Review by breeswish — APPROVED** [review]

File: util/logutil/log_test.go (hunk 3 above)
Comment (SabaPing) [reply util/logutil/log_test.go:131]: fixed

**Review by breeswish — APPROVED** [review]

File: util/logutil/log.go (hunk 2 above)
Comment (SabaPing) [reply util/logutil/log.go]: fixed

File: util/logutil/log.go [hunk 4]
```diff
@@ -106,20 +104,42 @@ func InitLogger(cfg *LogConfig) error {
...
+	var prop *log.ZapProperties
 	if len(os.Getenv("GRPC_DEBUG")) > 0 {
-		// more information for verbosity: https://github.com/google/glog#verbose-logging
-		gzap.ReplaceGrpcLoggerV2WithVerbosity(gl, GRPCLogDebugVerbosity)
+		config.Level = "debug"
+		l, prop, err = log.InitLogger(&config, zap.AddStacktrace(zapcore.FatalLevel))
+		if err != nil {
+			return nil, nil, errors.Trace(err)
+		}
+		gzap.ReplaceGrpcLoggerV2WithVerbosity(l, 999)
```
[the remark this answers is not in the document]
Comment (SabaPing) [reply util/logutil/log.go:132]: https://github.com/pingcap/tidb/blob/4f31cb49f083870f6b8e0f1ecc9183bc40c4ba42/tidb-server/main.go#L593

The original logic set the verbosity to 999. I just migrated the code and kept unchanged.

**Comment by breeswish** [comment]
@youjiali1995 @zimulala PTAL

File: util/logutil/slow_query_logger.go [hunk 5]
```diff
@@ -13,11 +13,11 @@ import (
 
 var _pool = buffer.NewPool()
 
-func newSlowQueryLogger(cfg *LogConfig) (*zap.Logger, error) {
+func newSlowQueryLogger(cfg *LogConfig) (*zap.Logger, *log.ZapProperties, error) {
```
Comment (zimulala) [inline util/logutil/slow_query_logger.go:16]: Why add the `*log.ZapProperties`? It seems that we don't use it.

File: util/logutil/log.go [hunk 6]
```diff
@@ -106,20 +104,42 @@ func InitLogger(cfg *LogConfig) error {
...
+	_, _, err = initGRPCLogger(cfg)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+func initGRPCLogger(cfg *LogConfig) (*zap.Logger, *log.ZapProperties, error) {
+	// copy Config struct by assignment
```
Comment (zimulala) [inline util/logutil/log.go:121]:
```suggestion
	// Copy Config struct by assignment.
```

**Comment by SabaPing** [comment]
I'll pick up this issue tomorrow.

File: util/logutil/slow_query_logger.go (hunk 5 above)
Comment (SabaPing) [reply util/logutil/slow_query_logger.go:16]: Those props are used in `TestSlowQueryLoggerCreation` function.

**Comment by SabaPing** [comment]
PTAL @tisonkun @breeswish @zimulala

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**